awx/main/models/jobs.py

```diff
@@ -350,66 +341,72 @@ def can_start_without_user_input(self, callback_extra_vars=None):
         # that of job template launch, so prompting_needed should
         # not block a provisioning callback from creating/launching jobs.
         if callback_extra_vars is None:
-            for value in self._ask_for_vars_dict().values():
-                if value:
+            for ask_field_name in set(self.ask_mapping.values()):
+                if getattr(self, ask_field_name):
                     prompting_needed = True
+                    break
         return (not prompting_needed and
                 not self.passwords_needed_to_start and
                 not variables_needed)
 
-    def _ask_for_vars_dict(self):
```

Comment:
I really like that we're replacing this with something dynamic, and specifying exceptions *at the place* the field is defined 👍